api/client/client.go

```diff
@@ -24,6 +26,8 @@ import (
 // A server points to a single remote CFSSL instance.
 type server struct {
 	URL string
+	RemoteCAs *x509.CertPool
```

Comment:
I'd prefer to just pass a full `*tls.Config` around, with the `Certificates` and `RootCAs` contained.